src/trace/context/extractors/step-function.spec.ts

```diff
@@ -42,6 +65,22 @@ describe("StepFunctionEventTraceExtractor", () => {
       expect(traceContext?.source).toBe("event");
     });
 
+    it("extracts trace context with valid redriven payload", () => {
+      // Mimick TraceContextService.extract initialization
+      StepFunctionContextService.instance(redrivePayload);
+
+      const extractor = new StepFunctionEventTraceExtractor();
+
+      // Payload is sent again for safety in case the instance wasn't previously initialized
+      const traceContext = extractor.extract(redrivePayload);
+      expect(traceContext).not.toBeNull();
+
+      expect(traceContext?.toTraceId()).toBe("1139193989631387307");
+      expect(traceContext?.toSpanId()).toBe("3615526381891637131");
```

Comment:
Could you link the PR line in the reducer that matches this test?

Also, is the plan to match traceID only but not spanID?

Reply:
Here's the [matching line](https://github.com/DataDog/logs-backend/blob/c5ae58dc4f823d2a74b4cba4b01b878cb4e2d72e/domains/serverless/apps/logs-to-traces-reducer/src/main/java/com/dd/logs/serverless/logs_to_traces/reducer/output/stepfunction/trace/spanbuilder/BaseSpanBuilder.java#L173) from logs-backend

Yes the lambda span should be in the same trace but we want the spanID to be different from the original when redrive count was 0

Reply:
Can you link the test just to verify that the two matches? Thanks!